ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/AgentTool.java

```diff
@@ -32,7 +32,7 @@
  */
 @Log4j2
 @ToolAnnotation(AgentTool.TYPE)
-public class AgentTool implements Tool {
+public class AgentTool implements WithoutModelTool {
```

Comment:
I would like to think the other way. Regardless of having model, any tool developer can extend the Tool interface. So let's not change the Tool interface. Instead we can extend the Tool interface to `ToolWithModel` which can have this extra method and the tools which aren't using models can implement Tool and the tools which are using models can implement `ToolWithModel` interface?

Reply:
Sure, Please check it. Also with the logic in AgentModelSearcher to construct query.

Reply:
But I have concern. For this, we should make sure when a new tool with mlmodel is developed, it needs to implement  `ToolWithModel` but not `Tool`. It's a requirement for code review and it's more like a soft rule. So I'm afraid in the future it may skip this logic if we code reviewers miss this.